tests/everflow/everflow_test_utilities.py

```diff
@@ -602,20 +602,67 @@ def _get_random_src_port(self, setup):
 
     def _get_monitor_port(self, setup, mirror_session, duthost):
         mirror_output = duthost.command("show mirror_session")
-        logging.info("Running mirror session configuration:\n%s", mirror_output["stdout"])
+        logging.info("mirror session configuration: %s", mirror_output["stdout"])
 
-        matching_session = list(filter(lambda line: line.startswith(mirror_session["session_name"]),
-                                       mirror_output["stdout_lines"]))
-        pytest_assert(matching_session, "Test mirror session {} not found".format(mirror_session["session_name"]))
-        logging.info("Found mirror session:\n%s", matching_session[0])
+        pytest_assert(mirror_session["session_name"] in mirror_output["stdout"],
+                      "Test mirror session {} not found".format(mirror_session["session_name"]))
+
+        sessions = {}
+        lns = mirror_output["stdout"].split("\n")
+        while True:
+            obj, lns = self._objectize(lns)
+            if obj == None:
+                break
+            sessions[obj["name"]] = obj
```

Comment:
This part was really hard to follow, what I would recommend is:

1) Move this loop into a helper called `parse_mirror_session_running_config` or something similar so 1) it's easier for others to re-use (`objectize` is helpful but most people will want the functionality provided by this loop as well if they are parsing mirror session data) and 2) it's more clear what this step is accomplishing in the context of `get_monitor_port`.
2) Re-name `objectize` to `parse_mirror_session_group` or something similar. My biggest confusion looking over this code is that I wasn't sure exactly what part of the output `objectize` was operating on/what data it was returning.

Alternatively, you could provide a helper method like `objectize` that accepts an argument for the type of mirror session you are looking for (ERSPAN or SPAN, at this point), and then it would be even more clear what data you're getting back when you call the method.